lib/client-assets.php

```diff
@@ -710,28 +710,28 @@ function gutenberg_register_scripts_and_styles() {
  * data to be attached to the page. Expected to be called in the context of
  * `array_reduce`.
  *
- * @param  array  $memo Reduce accumulator.
- * @param  string $path REST API path to preload.
- * @return array        Modified reduce accumulator.
+ * @param  array $memo Reduce accumulator.
+ * @param  array $args REST API path and method to preload.
+ * @return array       Modified reduce accumulator.
  */
-function gutenberg_preload_api_request( $memo, $path ) {
+function gutenberg_preload_api_request( $memo, $args ) {
 
 	// array_reduce() doesn't support passing an array in PHP 5.2
 	// so we need to make sure we start with one.
 	if ( ! is_array( $memo ) ) {
 		$memo = array();
 	}
 
-	if ( empty( $path ) ) {
+	if ( empty( $args ) || empty( $args['path'] ) || empty( $args['method'] ) ) {
```

Comment:
*Comment from old PR*
https://github.com/WordPress/gutenberg/pull/4155/files#r201324762

Aside: GET will certainly be the majority usage here. I'd expect it to be reasonable as a default value. Maybe even to the extreme (i.e. debatable) of supporting mixed value arrays like...
```
[
	'/get/to/path/1',
	'/get/to/path/2',
	[
		'path'   => '/options/to/path',
		'method' => 'OPTIONS',
	],
]
```
But this is less part of a public API than I originally expected, so I don't feel strongly about it at this point.

Reply:
I'm not a fan of the idea of mixing types in the array. I feel like at least this way it's obvious what's happening.

I would maybe say that this could throw an error if the args are not provided, at the moment it just skips preloading the request.

Also not sure if we'd consider splitting some code out of client-assets, we might be able to make this a bit more expressive.